x/rollapp/keeper/fraud_proposal.go

```diff
@@ -14,7 +14,7 @@ import (
 // HandleFraud handles the fraud evidence submitted by the user.
 func (k Keeper) HandleFraud(ctx sdk.Context, rollappID, clientId string, height uint64, seqAddr string) error {
 	// Get the rollapp from the store
-	_, found := k.GetRollapp(ctx, rollappID)
+	rollapp, found := k.GetRollapp(ctx, rollappID)
```

Comment:
what if a rollapp is already frozen? prob want to error on that.

Reply:
what happens if fraud submitted for height 100, accepted
but there's a fraud on height 90 as well, which is submitted later?

Reply:
good point! seems that currently when iterating over states and finalizing them we simply check if the rollapp is frozen or not. 
in order to suport this use case we should probably also change to check at which height it's frozen and only mark those states as "REVERTED" and the rest as finalized (even if the rollapp is frozen) as we're saying it's frozen since specific height. 
need to make sure also that the IBC client is frozen from that height. 

So I'd say for your question check if frozen, what's the frozen height and only accepts disputes with height < frozen height.